civic_scraper/platforms/granicus/site.py

```diff
@@ -0,0 +1,68 @@
+import civic_scraper
+import feedparser
+
+from civic_scraper import base
+from civic_scraper.base.asset import Asset, AssetCollection
+from civic_scraper.base.cache import Cache
+from datetime import datetime
+from requests import Session
+from pathlib import Path
+from urllib.parse import urlparse, parse_qs
+
+
+class GranicusSite(base.Site):
+    def __init__(self, rss_url, cache=Cache()):
+        self.url = rss_url
+        self.granicus_instance = urlparse(rss_url).netloc.split('.')[0]
+        self.cache = cache
+
+    def create_asset(self, entry):
+        asset_name = entry['title']
+        committee_name, asset_type, str_datetime = asset_name.split(' - ')
+        meeting_datetime = datetime.strptime(str_datetime, '%b %d, %Y %I:%M %p')
+
+        meeting_url = entry['link']
+        query_dict = parse_qs(urlparse(meeting_url).query)
+        try:
+            if 'ID' in query_dict.keys():
+                meeting_id = 'granicus_{}_{}'.format(self.granicus_instance, query_dict['ID'][0])
+            else:
```

Comment:
this should be moved to the config

Reply:
this addresses the fact that on a single granicus site, entries might have different query param names for the entry ID. I've added a comment here to clarify why this conditional is here

Reply:
okay, thanks!